src/preprocess/audio_waveform/preprocess.py

```diff
@@ -0,0 +1,147 @@
+import os
+
+from audiomentations import Compose, AddGaussianNoise, PitchShift, Reverse, Shift, TimeStretch
+import librosa
+import numpy as np
+
+from src.plotting.plot_audio_wav_augmentation import plot_waveforms
+
+np.random.seed(42)
+
+
+def truncate_or_fill(input_waveform: np.ndarray, samplerate: int, target_waveform_duration: float) -> np.ndarray:
+    """
+    :param input_waveform: np.ndarray, the alraedy loaded waveform as a numpy array.
+    :param samplerate: int, the sample rate of the waveform.
+    :param target_waveform_duration: float, the duration all waveforms should have.
+
+    :return truncated_waveform or filled_waveform: np.ndarray, the truncated or filled waveform.
+
+    Either truncates the wavefrom when it is longer than the target duration or fills the waveform if it is shorter.
+    Filling is done by repeating the waveform.
+    """
+    target_waveform_length = target_waveform_duration * samplerate
+
+    if len(input_waveform) >= target_waveform_length:
+        # truncate
+        truncated_waveform = input_waveform[0:target_waveform_length]
+        return truncated_waveform
+    else:
+        # fill
+        # divide target by current length and add 1 to make sure that the new length is higher than the target length
+        repitition_times = int(target_waveform_length/len(input_waveform) + 1)
+        repeated_waveform = np.repeat(input_waveform, repitition_times)
+
+        filled_waveform = repeated_waveform[0:target_waveform_length]
+        return filled_waveform
+
+
+def _add_gaussian_noise(input_waveform: np.ndarray, samplerate: int) -> np.ndarray:
+    """
+    :param input_waveform: np.ndarray, the already loaded waveform as a numpy array.
+    :param samplerate: int, the sample rate of the waveform.
+
+    :return transformed_wav: np.ndarray, the transfomred waveform.
+
+    Adds gaussain noise onto the waveform.
+    """
+    augment_with_gaussian_noise = Compose([
+        AddGaussianNoise(min_amplitude=0.005, max_amplitude=0.015, p=1.0)
+    ])
+    transformed_wav = augment_with_gaussian_noise(samples=input_waveform, sample_rate=samplerate)
+
+    return transformed_wav
+
+
+def _pitch_shift(input_waveform: np.ndarray, samplerate: int) -> np.ndarray:
+    """
+    :param input_waveform: np.ndarray, the already loaded waveform as a numpy array.
+    :param samplerate: int, the sample rate of the waveform.
+
+    :return transformed_wav: np.ndarray, the transfomred waveform.
+
+    Applies a pitch to the waveform.
+    """
+    augment_with_pitch_shift = Compose([
+        PitchShift(min_semitones=-4, max_semitones=4, p=1.0)
```

Comment:
magic numbers?

Reply:
Yes, magic numbers directly from the library's documentation. So I would use htem for nw and dive deeper into it if we are not satisfied with the performance